product_multi_company/models/product.py

```diff
@@ -1,19 +0,0 @@
-# Copyright 2023 Karthik <karthik@sodexis.com>
-# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
-
-from odoo import fields, models
-
-
-class ProductProduct(models.Model):
-    _inherit = "product.product"
-
-    company_ids = fields.Many2many(
```

Comment:
Why removing this?

Reply:
It was not in v15, it is not necessary

Reply:
The fact is that by extending the abstract model in product.template, product.product acquires that field.

Reply:
@pilarvargas-tecnativa 
From what I know, the above concept is the reason for the user A and user B issue.
As product.product doesn't hold the company_ids directly in it, if someone tries to access the product.product then it tries to compute it from the product.template.
This causing the above issue as the user doesn't have access to all the companies that is set on the product.template.